lib/node_modules/@stdlib/blas/base/idamax/src/idamax.c

```diff
@@ -0,0 +1,65 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/base/idamax.h"
+#include <math.h>
+
+/**
+* Finds the index of the first element having maximum fabsolute value.
+*
+* @param N       number of indexed elements
+* @param X       input array
+* @param strideX  index increment for `x`
+*/
+int c_idamax( const int N, const double *X, const int strideX ) {
+	double dmax;
+	int idx;
+	int ix;
+	int i;
+
+	idx = 0;
+	if ( N < 1 || strideX <= 0 ) {
+		return -1;
+	}
+	if ( N == 1 ) {
+		return idx;
+	}
+	if (strideX == 1 ) {
+		// Code for stride equal to `1`...
+		dmax = fabs( X[ 0 ] );
```

Comment:
We should use `stdlib_base_abs`, no?

Reply:
currently, the tooling needs to be adjusted to allow stdlib functions to used directly in the C implementation. The dasum BLAS routine also has a similar format.

Reply:
This has been resolved upstream. I'll make the corresponding changes to this PR.